tests/gflownet/envs/common.py

```diff
@@ -223,14 +226,19 @@ def test__step__returns_same_state_action_and_invalid_if_done(env):
         policy_outputs=random_policy, mask_invalid_actions=mask_invalid
     )
     action = actions[0]
-    env.set_state(env.state, done=True)
-    next_state, action_step, valid = env.step(action)
-    if torch.is_tensor(env.state):
-        assert torch.equal(next_state, env.state)
-    else:
-        assert next_state == env.state
-    assert action_step == action
-    assert valid is False
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        env.set_state(env.state, done=True)
+    if env.done:
```

Comment:
So this test now doesn't do anything for environments where the initial state cannot be done (like crystal and composition)? Should this state be modified so that it will actually test something on all envs?
I think we could modify the test to sample actions randomly from the mask of valid actions until we reach a state where we can do `env.set_state(env.state, done=True)` and `env.done` becomes True. And then we can do the rest of the test. This way, I think it will work for all environments. What do you think?

Reply:
I had thought of doing this but was lazy about it :sweat_smile: Since you said it too, I have done it. 

While doing that, I thought of implementing a couple of helper functions to sample random steps and trajectories, which I have used in other tests, so the PR has grown a little. Sorry for the mess.